packages/template-retail-react-app/app/pages/product-detail/index.jsx

```
@@ -413,30 +408,22 @@ const ProductDetail = () => {
                 }
             }
 
-            // Set target to default shipment
-            let targetShipmentId = 'me'
-
-            if (basket) {
-                // Ensure a suitable shipment exists
-                if (hasAnyPickupSelected) {
-                    targetShipmentId = await findOrCreatePickupShipment(productItems, selectedStore)
-                } else {
-                    targetShipmentId = await findOrCreateDeliveryShipment(
-                        productItems,
-                        selectedStore
-                    )
-                }
-
-                if (!targetShipmentId) {
-                    throw new Error('Failed to find or create shipment')
-                }
+            // Fetch and assign a suitable shipment for product items
+            const targetShipmentId = await getShipmentForItems(
+                productItems,
+                selectedStore,
+                hasAnyPickupSelected
+            )
 
-                productItems = productItems.map((item) => ({
-                    ...item,
-                    shipmentId: targetShipmentId
-                }))
+            if (!targetShipmentId) {
+                throw new Error('Failed to find or create shipment')
             }
```

I'm not convinced we should surface this error message to the shopper, since it doesn't really mean anything to them. My preference would be to throw standard api error ("Something went wrong") as a toast, but I'm not attached to either. (Note - I haven't intl this text yet due to this)

---

as coded it's impossible to hit this error, i think.  that said, maybe we ignore the null return case and instead wrap the map below in `if (targetShipmentId) {` ?